library/Fission/User/Creator/Class.hs

```diff
@@ -0,0 +1,72 @@
+module Fission.User.Creator.Class
+  ( Creator (..)
+  , Errors
+  ) where
+
+import           Data.UUID (UUID)
+import           Database.Esqueleto
+
+import           Fission.Models
+import           Fission.Prelude
+
+import qualified Fission.Platform.Heroku.Region.Types  as Heroku
+import qualified Fission.Platform.Heroku.AddOn.Creator as Heroku.AddOn
+
+import qualified Fission.User.Creator.Error as User
+import qualified Fission.User.Password      as Password
+import           Fission.User.Role.Types
+
+type Errors = OpenUnion
+  '[ User.AlreadyExists
+   , Heroku.AddOn.Error
+   , Password.FailedDigest
+   ]
```

Comment:
This is an example of `OpenUnion` in the wild

Reply:
oh yeah I dig this quite a bit